### ISSUE TITLE
Mana not lost when failing spell as priest or paladin

### ISSUE THREAD

**Issue by supersnail12345** [dungeons-of-moria/umoria#29]
When casting priest spells mana is not lost on a failed casting. Issue tested on v 5.7.10 and 5.7.11, Windows build, moria (non-rogue) keyset. Problem can easily be reproduced using wizard mode and attempting to cast high-fail rate spells such as holy word. This also affects paladins. Mage spells                              are not affected by this bug, and subtract the mana cost on a failed casting per expected behavior.

**Comment by mrcook** [comment]
Thanks for the detailed report Saru, that will help a lot in testing this. I'll take a look as soon as I can.

**Comment by mrcook** [comment]
Hi @supersnail12345, I've now found and fixed the bug for this: https://github.com/dungeons-of-moria/umoria/commit/1a14de308b304ba15dce067fe879aa94de349bea

It seems that back when I first started cleaning up the original 5.6.0 source code I had not correctly understood the tricky if/else logic around this part of the code. Hopefully this is the only instance of this kind of error!

Thanks again for reporting the issue.
Michael
